core/base/src/test/java/org/eclipse/dataspaceconnector/core/CoreServicesExtensionTest.java

```diff
@@ -16,18 +16,18 @@
 
 import org.eclipse.dataspaceconnector.policy.model.PolicyRegistrationTypes;
 import org.eclipse.dataspaceconnector.spi.security.PrivateKeyResolver;
+import org.eclipse.dataspaceconnector.spi.system.ExecutorInstrumentation;
 import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.system.injection.ObjectFactory;
 import org.eclipse.dataspaceconnector.spi.types.TypeManager;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-import static org.mockito.ArgumentMatchers.anyInt;
-import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
+@ExtendWith(TypeManagerDependencyInjectionExtension.class)
```

Comment:
This starts to be overcomplicated for a test that verifies an initialization and not a behavior. I would remove the test (`verifyPolicyTypesAreRegistered`) as it doesn't give significant value

Reply:
we do test serialization (`PolicyTest`), but it's not just about making sure that Policies and all the other types can be deserialized, it's also about making sure that we ship code that has those types _already registered_. 

Otherwise we could run into a situation where serialization per se works fine, but the API still crashes, because we forgot to register some `Constraint` implementor with the `TypeManager`.